magpie/run/magpie-run-project-hive

```diff
@@ -0,0 +1,219 @@
+#!/bin/bash
+#############################################################################
+#  Copyright (C) 2013-2018 Army Research Laboratory (ARL) .
+#  Produced at Army Research Laboratory (cf, DISCLAIMER).
+#  Written by Adam Childs <adam.s.childs.ctr@mail.mil>
+#  LLNL-CODE-644248
+#  
+#  This file is part of Magpie, scripts for running Hadoop on
+#  traditional HPC systems.  For details, see https://github.com/llnl/magpie.
+#  
+#  Magpie is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#  
+#  Magpie is distributed in the hope that it will be useful, but
+#  WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with Magpie.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+# These are functions to be called by magpie-run
+
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-submission-type
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-dirs
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-user
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-defaults
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-log
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-paths
+
+# Return 0 if service up, 1 if not
+__Magpie_run_check_hive_up () {
+    cd ${HIVE_HOME}
+
+    local serverschk=0
+    local serversup=0
+    for server in `cat ${HIVE_CONF_DIR}/masters`
+    do
+        serverschk=`expr ${serverschk} + 1`
+        local serverisup=`${MAGPIE_REMOTE_CMD:-ssh} ${MAGPIE_REMOTE_CMD_OPTS} $server ps aux | grep '[H]iveServer2' | awk '{print $2}'`
+        if [ "${serverisup}" -gt "0" ]
+        then
+            serversup=`expr ${serversup} + 1`
+        fi
+    done
+    
+    echo "$serversup/$serverschk Hive Servers are up."
+    if [ "$serverschk" -eq "$serversup" ]
+    then
+        return 0
+    fi
+
+    return 1
+}
+
+Magpie_run_start_hive () {
+    if [ "${HIVE_SETUP}" == "yes" ] && [ "${magpie_run_prior_startup_successful}" == "true" ]
+    then
+        if [ "${magpie_run_hadoop_setup_successful}" == "0" ]
+        then
+            Magpie_output_internal_error "Attempt to setup Hive without Hadoop being setup"
+            magpie_run_hive_should_be_torndown=0
+            magpie_run_hive_setup_successful=0
+            return 1
+        fi
+
+        cd ${HIVE_HOME}
+
+        if [ ${HIVE_MODE} != "setuponly" ]
+        then
+
+            Magpie_make_all_local_dirs_node_specific
+            echo "Starting Postgres daemon"
+            ${MAGPIE_SCRIPTS_HOME}/bin/magpie-postgres-daemon.sh start ${HIVE_CONF_DIR}
+
```

Comment:
do we have to wait for the postgres daemon to be up?  Perhaps do a "is postgres up" check?

Reply:
We do check in the magpie-hive-daemon.sh on line 84...would you prefer that be moved into this file or remain in the daemon?

Reply:
I think for consistency it should be in this file.  Because everywhere else in Magpie the "is it up yet" checks are in here.